contracts/MarginCalculator.sol

```diff
@@ -350,4 +350,51 @@ contract MarginCalculator is Initializable {
     function _isEmptyAssetArray(address[] memory _assets) internal pure returns (bool) {
         return _assets.length == 0 || _assets[0] == address(0);
     }
+
+    /**
+     * @dev convert a uint256 amount
+     * Examples:
+     * (1)  USDC    decimals = 6
+     *      Input:  8000000 USDC =>     Output: 8 * 1e18 (8.0 USDC)
+     * (2)  cUSDC   decimals = 8
+     *      Input:  8000000 cUSDC =>    Output: 8 * 1e16 (0.08 cUSDC)
+     * @return internal amount that is sacled by 1e18.
+     */
+
+    function _tokenAmountToInernalAmount(uint256 _amount, address _token) internal view returns (uint256) {
+        ERC20Interface token = ERC20Interface(_token);
+        uint256 decimals = uint256(token.decimals());
+        uint256 base = 18;
+        if (decimals == base) return _amount;
+        if (decimals > base) {
+            uint256 exp = decimals - base;
+            return _amount.div(10**exp);
```

Comment:
isn't this a safe math div? So doesn't this do 

8000000 / 10^6? Where is the scaling by 1e18 happening?

Reply:
Also you should probs first multiply by 1e18, since if you pass in something like 80 for USDC, it will show 0 in internal values right?

Reply:
this case is if the USDC decimals is higher than 18. Let's say there's a token has 20 decimals and you pass in 20, then yes it will be truncated and the output will be 0.

Reply:
if the token has decimals 20, it already scaled by 1e20 in our definition, we just have to remove the last 2 digits to get a number scaled in 1e18.